### PR TITLE
#2325-Refactor-frontend

### PR DESCRIPTION
**Pull request by somme**

### CODE REVIEW

**Comment by JazzPote** [comment]
Nice one! Looks fresh

Can we:
- Move scss/ to frontend/
- Have a util/ folder in each backend and frontend, and move each util module accordingly
- move app.ts and server.ts to backend/
- move public/ under src/


What do you think

File: src/backend/app.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        }
+      )
+    )
+
+    this.app.get('/login', passport.authenticate('oauth2'))
+    this.app.get(
+      '/auth/gds/callback',
+      passport.authenticate('oauth2', '/error-callback'),
+      async (req, res) => res.redirect('/')
+    )
```
Comment (JazzPote) [inline src/backend/app.ts:138]: With these being technically routes, how about having a `LoginRoute` or `AuthRoute` class?
The /reauth endpoint will be there and will need a controller.

Comment (JazzPote) [reply src/backend/app.ts:138]: Happy to address this change when rebasing my PR on this one

**Review by JazzPote — COMMENTED** [review]
A couple more nit picks

File: src/backend/app.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+const redisStore = new RedisStore({
+  client: redisInstance,
+  prefix: 'GovSearch::',
+})
+
+const views = [
+  path.join(__dirname, '../../node_modules/govuk-frontend'),
+  path.join(__dirname, './views'),
+]
```
[the remark this answers is not in the document]
Comment (somme) [reply src/backend/app.ts]: added

File: src/backend/app.ts (hunk 1 above)
Comment (somme) [reply src/backend/app.ts:135]: Happy for you do address this as your PR is to do with Auth

**Review by JazzPote — APPROVED** [review]
LGTM - Nice update!

**Pull request merged**